### ISSUE TITLE
Required pre/post processing for MediaPipe selfie multi-class on stand-alone TensorFlow or ONNX

### ISSUE THREAD

**Issue by ali4864** [google-ai-edge/mediapipe#6134]
I can have an expected segmentation by using MediaPipe APIs (selfie multi-class) but I can't get the same result by running the model on stand-alone (without using MediaPipe APIs) TensorFlow or ONNX (converted TFLite to ONNX model)! I think it is because of some missing steps before feeding data (image) for inference or some missing step for post processing of model outputs. Are there any required pre/post processing steps? for example I know that I need to applying softmax on model outputs.

**Comment by HarishPermal** [comment]
Hi @ali4864,

Thanks for brining this up, I needed some time to check with my developer team on this.

**Comment by HarishPermal** [comment]
Hi @ali4864,

Apologies for the delayed response, To replicate mediaPipe's results you need to convert the image to RGB. Libraries like OpenCV often load images in BGR format so conversion is necessary as the models are typically trained on RGB data. Then for the selfie multiclass model resize the image to exact input tensor 256X256 pixels. And the input data to the type float32.

For postprocessing, apply a softmax along the last axis then apply an argmax operation along the last axis of the softmax output. This will result in a tensor of shape (1, 256, 256). Then resize the Argmax output back to the original input images dimensions.

Hope this helps

**Comment by ali4864** [comment]
Hi @HarishPermal

I'm so sorry for the late response, recently I'm handling a high volume of tasks. Thank you very much for your assistance. I will test your solution and I will share the result.
